WalletWasabi/Userfacing/AddressStringParser.cs

```diff
@@ -36,48 +36,82 @@ public static bool TryParseBitcoinAddress(string text, Network expectedNetwork,
 	}
 
 	public static bool TryParseBitcoinUrl(string text, Network expectedNetwork, [NotNullWhen(true)] out BitcoinUrlBuilder? url)
+		=> TryParseBitcoinUrl(text, expectedNetwork, out _, out url);
+
+	public static bool TryParseBitcoinUrl(string text, Network expectedNetwork, [NotNullWhen(false)] out string? errorMessage, [NotNullWhen(true)] out BitcoinUrlBuilder? url)
 	{
+		errorMessage = null;
 		url = null;
 
 		if (text is null || expectedNetwork is null)
 		{
+			errorMessage = "Internal error.";
 			return false;
 		}
 
 		text = text.Trim();
 
 		if (text.Length is > 1000 or < 20)
 		{
+			errorMessage = "Input length is invalid.";
 			return false;
 		}
 
-		try
+		if (!text.StartsWith("bitcoin:", true, CultureInfo.InvariantCulture))
 		{
-			if (!text.StartsWith("bitcoin:", true, CultureInfo.InvariantCulture))
-			{
-				return false;
-			}
+			errorMessage = "'bitcoin:' prefix is missing.";
+			return false;
+		}
```

Comment:
Showing an error message should be the last option. First, if possible we should fix the problem automatically without bothering the user.
In this case, if the prefix is missing, but the remaining content is valid, just add the prefix.